hacker_news_to_sqlite/cli.py

```diff
@@ -124,6 +124,15 @@ def ensure_tables(db):
             {"id": int, "type": str, "by": str, "time": int, "title": str, "text": str},
             pk="id",
         )
+    # includes hidden columns
+    all_column_names = {
+        c[1] for c in db.execute("PRAGMA table_xinfo([items])").fetchall()
+    }
+    if "permalink" not in all_column_names:
+        db.execute(
```

Comment:
Could be improved by a resolution on https://github.com/simonw/sqlite-utils/issues/411

Reply:
Also I was mulling this over a bit - making the table virtual provides backwards compatibility, but it might be better to create a real column for new databases. It'll take up space storing a lot of nearly-identical strings, but it won't incur repeated computation at runtime.

I was thinking about it in the context of adding a `num_children` column to make interacting with the string `kids` column easier. This can be done in pure sqlite as another virtual column, but it would be easier to pre-compute it in new databases (and provide a virtual table for existing ones)